### PR TITLE
🧹 Pass arguments with 'in' explicitly where applicable

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.LocalFunctions.cs [hunk 1]
```diff
@@ -191,20 +191,20 @@ protected override LocalFunctionState LocalFunctionStart(LocalFunctionState star
                 currentState.InvertedCapturedMask.Invert();
             }
             // Filter the modified state variables to only captured variables
-            stateAtReturn.Assigned.IntersectWith(currentState.CapturedMask);
+            stateAtReturn.Assigned.IntersectWith(in currentState.CapturedMask);
```
Comment (mavasani) [inline src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.LocalFunctions.cs:194]: Were all the changes done manually or through some analyzer/code fixer?

**Comment by gafter** [comment]
What is the motivation for making this change?

File: src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.LocalFunctions.cs (hunk 1 above)
Comment (sharwell) [reply src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.LocalFunctions.cs:194]: The changes were automated through a code fix.

**Comment by sharwell** [comment]
> What is the motivation for making this change?

`in` parameters presumably are written to avoid copying large value types, but the compiler still allows a copy to be made at the call site in various cases. Explicitly specifying `in` disallows the copy that would otherwise be possible, which makes it easier to understand the performance characteristics of code.

**Comment by gafter** [comment]
@dotnet/roslyn-compiler Do any of you have any concern with this systematic change?

**Comment by sharwell** [comment]
As a bit of background, I'm using an analyzer to eliminate a large number of large struct copies in the formatter implementation. This pull request just cleans up many remaining cases it found.

Eventually I'd like to bring this in as a code style analyzer, but there are a few bugs in it that need to be addressed before that's a real option.

**Comment by sharwell** [comment]
Superseded by #44197 (same change, but brings analyzer enforcement)

**Pull request closed without merging**